packages/sdk/src/TemplateProvider.ts

```diff
@@ -6,51 +6,91 @@ import * as fs from 'fs'
 import * as path from 'path'
 import { Template, TemplateVariable, RenderedActionArgument } from '@conversationlearner/models'
 import { CLDebug } from './CLDebug'
+import { TemplateEngine, ActivityFactory } from 'botbuilder-lg'
 
 /**
- * Provider for rendering templates
+ * Provider for rendering LG templates.
  */
 export class TemplateProvider {
+
     private static hasSubmitError = false
 
+    public static LGTemplateDirectory(): string | null {
+        const cwd = process.cwd()
+        const lgDirectoryName = "lg"
+        // TODO - make this configurable.
+        let templateDirectory = path.join(cwd, `./${lgDirectoryName}`)
+        if (fs.existsSync(templateDirectory)) {
+            return templateDirectory
+        }
+        // Try up a directory or two as could be in /lib or /dist folder depending on deployment
+        templateDirectory = path.join(cwd, `../${lgDirectoryName}`)
+        if (fs.existsSync(templateDirectory)) {
+            return templateDirectory
+        }
+        templateDirectory = path.join(cwd, `../../${lgDirectoryName}`)
+        if (fs.existsSync(templateDirectory)) {
+            return templateDirectory
+        }
+        return null
+    }
+
     // TODO: Decouple template renderer from types from Action classes
     // E.g. use generic key,value object instead of RenderedActionArgument
-    public static async RenderTemplate(templateName: string, templateArguments: RenderedActionArgument[]): Promise<any | null> {
-        let template = this.GetTemplate(templateName)
-        if (template === null) {
+    public static async RenderTemplate(templateName: string, templateArguments: RenderedActionArgument[], entityDisplayValues: Map<string, string>): Promise<any | null> {
```

Comment:
Could be `Record` instead of `Map`. If I remember these entity value types were done when maps were just introduced to JavaScript and wasn't clear when to apply.

Reply:
I'd prefer to punt on this for this PR, since changing the method that populates this (`getEntityDisplayValueMap` in `FilledEntity`) would introduce several unrelated changes across Models, SDK, and UI.